packages/core/src/external/commonwell/document/document-downloader-local.ts

```diff
@@ -75,6 +79,58 @@ export class DocumentDownloaderLocal extends DocumentDownloader {
     return newlyDownloadedFile;
   }
 
+  /**
+   * Checks if the content type of a downloaded document is accepted. If not accepted, updates the content type
+   * and extension  in S3 and returns the updated download result.
+   */
+  async checkAndUpdateMimeType({
+    document,
+    fileInfo,
+    downloadedDocument,
+    downloadResult,
+  }: {
+    document: Document;
+    fileInfo: FileInfo;
+    downloadedDocument: string;
+    downloadResult: DownloadResult;
+  }): Promise<DownloadResult> {
+    if (isContentTypeAccepted(document.mimeType)) {
+      return { ...downloadResult };
+    }
+
+    const maxBytesNeeded = 6; //NOTE: if you update detectFileType, you might need to update this number
+    const partialBuffer = Buffer.from(downloadedDocument.slice(0, maxBytesNeeded));
+    let [detectedFileType, detectedExtension] = detectFileType(partialBuffer);
+
+    // If the detected file type is binary, check if it's likely a text file using the entire buffer.
+    if (detectedFileType === "application/octet-stream") {
+      if (isLikelyTextFile(Buffer.from(downloadedDocument))) {
+        detectedFileType = "text/plain";
+        detectedExtension = ".txt";
+      }
+    }
```

Comment:
this is a code smell - why don't we just put all of this code in `detectFileType()`?

Reply:
It was there, but it requires the whole file content on the Buffer. Since most files are not `.txt`, its cheaper to check the file type first based on the first few bytes and only copy the whole content to a Buffer if we can't determine from that.

Reply:
oh yeah for sure - that part makes sense

but then why don't we just have this check at the end of `detectFileType()` before the return?

will require a slightly diff structure in the func, but make the code cleaner & more encapsulated

Reply:
> why don't we just have this check at the end of detectFileType() before the return?

Yeah, could change the function to get the contents and create the Buffer inside it accordingly.

I had already iterated multiple times on this PR, didn't want to drag it too much, but I agree this is a better approach.